### DISCUSSION TITLE
Can JSBSim add a air disturbance when autopilot?

### DISCUSSION THREAD

**Discussion by theorynice** [JSBSim-Team/jsbsim#438] [Q&A]
JSBSim brings me great help. I have developed an autopilot controller in JSBSim, but I want to test its ability by adding some air disturbance. How can I do that?

**Comment by seanmcleod** [comment]
JSBSim has a couple of turbulence models available - 

https://github.com/JSBSim-Team/jsbsim/blob/375f5be0fdfeb210f4cf66f938e3e6fd426c2966/src/models/atmosphere/FGWinds.h#L56-L67

**Reply by theorynice** [reply]
Thank you. It helps me much!

**Reply by theorynice** [reply]
Excuse me, when I set the turbulence type as 3 (ttMilspec) and the other two parameters: 'atmosphere/turbulence/milspec/windspeed_at_20ft_AGL-fps' as 50
'atmosphere/turbulence/milspec/severity' as 3
The flight result shows it really works.
But when I plot the wind information, including 'atmosphere/psiw-rad' and 'atmosphere/wind-mag-fps', their values are invariant from beginning to end.   So, how can I show this turbulence signal?

**Reply by theorynice** [reply]
Sorry, it's my mistake, I should plot total wind information instead, rather than steady constant wind information.

**Comment by seanmcleod** [comment]
Read the documentation in FGWinds.h, i.e. continue reading below the snippet I posted above. There is more documentation regarding the options and even some examples.

**Comment by seanmcleod** [comment]
Take a look at:

https://github.com/JSBSim-Team/jsbsim/blob/375f5be0fdfeb210f4cf66f938e3e6fd426c2966/src/models/atmosphere/FGWinds.cpp#L561-L564
